app/login/index.js

```diff
@@ -0,0 +1,256 @@
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                username
+            })
+        }).then((res) => res.json().then((json) => {
+            if (res.status !== 200) {
+                this.setState({
+                    status: 'failure',
+                    statusMessasge: json.error.message
+                });
+                this.setState({loggedIn: false});
+            }
+            this.setState({loggedIn: json.approved});
+        })).catch((e) => {
+            this.setState({loggedIn: false});
+        });
+    }
+
+    authenticateUser () {
+        if (!this.state.username) {
+            this.setState({
+                status: 'failure',
+                statusMessasge: 'Enter your Plotly username.'
+            });
+            return;
+        }
+        if (!this.state.domain && this.state.serverType === ONPREM) {
+            this.setState({
+                status: 'failure',
+                statusMessasge: 'Enter your Plotly On Premise domain.'
+            });
+            return;
+        }
+        this.setState({statusMessasge: ''});
+        this.oauthPopUp();
+    }
+
+    logIn () {
+        const {username} = this.state;
+        this.verifyAuthDone();
+        if (!this.state.loggedIn) {
+            this.authenticateUser();
+            const checkAuth = setInterval(() => {
+                this.verifyAuthDone();
+                this.setState({
+                    statusMessasge:
+                    `Loading. User authorization of [${username}] in process.`
+                });
+                if (this.state.loggedIn) {
+                    clearInterval(checkAuth);
+                    window.location.assign(connectorUrl);
+                }
+            }, 1000);
+        } else {
+            window.location.assign(connectorUrl);
+        }
+    }
+
+    updateStateWithEvent(e) {
+        this.setState({[e.target.name]: e.target.value});
+    }
+
+    render() {
+        const renderOption = (value) => {
+            const selected = this.state.serverType === value;
+            return (
+                <span>
+                    <button
+                        className={selected ? 'button-primary' : 'button'}
+                        onClick={(e) => this.setState({serverType: e.target.value})}
+                        value={value}
+                        style={{margin: '10px'}}
+                    >{SERVER_TYPES[value]}</button>
+                </span>
+            );
+        };
+
+        const loginButton = (
+            <button
+                className="btn btn-large btn-primary"
+                style={{display: 'block', margin: 'auto'}}
+                onClick={() => this.logIn()}
+            >{'Login'}</button>
+        );
+
+        const serverTypeOptions = (
+            <div className="control-group">
+            <h3 className="block-center-heading">
+                Choose Your Plotly Use Case
```

Comment:
I still think that `Use Case` isn't the right term here. What if we did: `"I am connecting to"`

```
I am connecting to...

[Plotly Cloud] [Plotly On Premise]
```